src/plugins/home/server/services/tutorials_registry.ts

```diff
@@ -0,0 +1,88 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import Joi from 'joi';
+import { CoreSetup } from 'src/core/server';
+import {
+  TutorialProvider,
+  TutorialContextFactory,
+  ScopedTutorialContextFactory,
+} from '../lib/tutorials_registry_types';
+import { tutorialSchema } from '../lib/tutorial_schema';
+
+export class TutorialsRegistry {
+  private readonly tutorialProviders: TutorialProvider[] = []; // pre-register all the tutorials we know we want in here
+  private readonly scopedTutorialContextFactories: TutorialContextFactory[] = [];
+
+  public setup(core: CoreSetup) {
+    const router = core.http.createRouter();
+    router.get(
+      { path: '/api/kibana/home/tutorials', validate: false },
+      async (context, req, res) => {
+        const initialContext = {};
+        const scopedContext = this.scopedTutorialContextFactories.reduce(
+          (accumulatedContext, contextFactory) => {
+            return { ...accumulatedContext, ...contextFactory(req) };
+          },
+          initialContext
+        );
+
+        return res.ok({
+          body: this.tutorialProviders.map(tutorialProvider => {
+            return tutorialProvider(scopedContext); // All the tutorialProviders need to be refactored so that they don't need the server.
+          }),
+        });
+      }
+    );
+    return {
+      registerTutorial: (specProvider: TutorialProvider) => {
+        const emptyContext = {};
+        const { error } = Joi.validate(specProvider(emptyContext), tutorialSchema);
```

Comment:
NIT: `Joi.validate(specProvider({}), tutorialSchema);`?

Reply:
That's the code as it was prior to the migration.